GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -46,7 +51,11 @@ private void FormDeleteBranchLoad(object sender, EventArgs e)
             Branches.BranchesToSelect = Module.GetRefs(RefsFilter.Heads).ToList();
             foreach (var branch in Module.GetMergedBranches())
             {
-                if (!branch.StartsWith("* "))
+                if (branch.StartsWith("+ "))
+                {
+                    _worktreeBranches.Add(branch.Trim('+', ' '));
+                }
+                else if (!branch.StartsWith("* "))
                 {
                     _mergedBranches.Add(branch.Trim());
                 }
```

Comment:
Unfortunately, this approach does not suffice if the branch to be deleted has not been merged into the *current* checkout. First, I am forced to enable `force delete`. Then, only the git output gives the error message.

So, the check for worktrees should be done independently from "merged branches".

```
$ git branch --merged
* (HEAD detached at 6a15d4603)
  7795_process_start
  fix/6527_git_errors_prep2
  fix/toolbar_icon_switch
  fix_9104_stage-unstage_woes
  master
```
![grafik](https://user-images.githubusercontent.com/36601201/121587114-b9d84280-ca34-11eb-9551-00c6bad5df72.png)
![grafik](https://user-images.githubusercontent.com/36601201/121587261-deccb580-ca34-11eb-8c21-bab6d3acf0ae.png)

Reply:
If we are gonna check worktree branches independently maybe it is better to disable these branches during selection? There is no reason to let user try to delete them anyway.

Reply:
> If we are gonna check worktree branches independently maybe it is better to disable these branches during selection? There is no reason to let user try to delete them anyway.

👍
To be considered: If the dialog is entered via the context menu, the branch name is prepopulated and will not match.
=> error message (in singular)